### PR TITLE
refs #68; turn some macros to CT procs; fix #112 (doc fixes)

### LINKED ISSUE #112: `nim doc --project` is misused
`nim doc --project` is supposed to be for several modules at once (and generates a single theindex.html) so i dont' understand this:
```
  execCmd "nim doc --project --index:on nimterop/cimport"
  execCmd "nim doc --project --index:on nimterop/git"
  execCmd "nim doc --project --index:on nimterop/plugin"
```

instead shd probably be:
```
execCmd "nim doc --project --index:on nimterop/all" #which imports all  (in case there's not already such module importing `all`)
```

### PR DESCRIPTION
**Pull request by timotheecour**
* refs #68
* fixes https://github.com/nimterop/nimterop/issues/112
* run `nim doc` (including runnableExamples) as part of CI
* improved the way we're running `nim doc`
* various other fixes

### CODE REVIEW

File: tests/tnimterop_c.nim [hunk 1]
```diff
@@ -1,12 +1,13 @@
 import std/unittest
 import nimterop/cimport
 
-cDebug()
-cDisableCaching()
+static:
+  cDebug()
+  cDisableCaching()
+  cAddSearchDir "$projpath/include"
```
Comment (genotrance) [inline tests/tnimterop_c.nim]: testsIncludeDir()?

File: tests/tnimterop_c.nim [hunk 2]
```diff
@@ -137,7 +138,7 @@ var
   k: uKernel
   kp: Kernel
 
-cAddStdDir()
+static: cAddStdDir()
```
Comment (genotrance) [inline tests/tnimterop_c.nim]: Can be removed since it is being tested in tmath.

**Comment by genotrance** [comment]
I think we also need to mention somewhere that you have to use static or things won't work. I know it's in the examples but all the same.

File: nimterop/cimport.nim [hunk 3]
```diff
@@ -302,46 +299,32 @@ macro cIncludeDir*(dir: static string): untyped =
...
   runnableExamples:
-    cAddStdDir()
-    echo cSearchPath("math.h")
-
-  result = newNimNode(nnkStmtList)
-
+    static: cAddStdDir()
+    import os
+    doAssert cSearchPath("math.h").existsFile
   var
```
[the remark this answers is not in the document]
Comment (timotheecour) [reply nimterop/cimport.nim]: they're documenting unittests and the preferred way (in stdlib and most other projects) is to use doAssert (or assert), not echo; see https://nim-lang.github.io/Nim/contributing.html section on that.

Also, doAssert is the best documentation! concise and precise.
`echo cSearchPath("math.h").existsFile` tells me nothing when i read the docs
and something like `echo cSearchPath("math.h").existsFile # prints true` is not guaranteed to stay in sync, and more verbose

File: tests/tnimterop_c.nim (hunk 2 above)
Comment (timotheecour) [reply tests/tnimterop_c.nim]: done, good idea

File: nimterop/cimport.nim [hunk 4]
```diff
@@ -164,18 +164,15 @@ macro cOverride*(body): untyped =
...
   ## Similar to `cOverride() <cimport.html#cOverride.m,>`_, this macro allows
   ## filtering out symbols not of interest from the generated output.
-  ##
   runnableExamples:
-    cSkipSymbol "proc1", "Type2"
-
-  for skip in skips:
-    gStateCT.symOverride.add skip.strVal
+    static: cSkipSymbol @["proc1", "Type2"]
+  for a in skips: gStateCT.symOverride.add a
```
[the remark this answers is not in the document]
Comment (timotheecour) [reply nimterop/cimport.nim]: yup, done

**Comment by timotheecour** [comment]
> I think we also need to mention somewhere that you have to use static or things won't work. I know it's in the examples but all the same.

indeeed, I thought it would give an error but it doesn't, see https://github.com/nim-lang/Nim/issues/10535

will have to add documentation then

File: tests/tnimterop_c.nim (hunk 1 above)
Comment (timotheecour) [reply tests/tnimterop_c.nim]: thanks! replaced everywhere

File: nimterop/cimport.nim [hunk 5]
```diff
@@ -241,55 +238,55 @@ proc cSearchPath*(path: string): string {.compileTime.}=
...
-  ##
-  ##    cAddSearchDir("path/to/includes")
-  ##    cImport cSearchPath("file.h")
-
+  runnableExamples:
+    import paths, os
+    static:
+      cAddSearchDir testsIncludeDir()
+    doAssert cSearchPath("test.h").existsFile
   var dir = interpPath(dir)
```
[the remark this answers is not in the document]
Comment (timotheecour) [reply nimterop/cimport.nim]: I added commits that run `nim doc` as part of CI, and they're green, so resolving this :)

**Pull request merged**